src/utils/emails.py

```diff
@@ -111,8 +136,7 @@ def send_mailgun(
             auth=("api", self.sending_profile["mailgun_api_key"]),
             data=data,
             files=[
-                ("attachment", ("report.pdf", open(a, "rb").read()))
-                for a in attachments
+                ("attachment", (filename, open(a, "rb").read())) for a in attachments
```

Comment:
Does this close the file after opening it? Normally I would use a context manager to automatically close files after reading them

Reply:
@xlyk very good point. I'll refactor this portion